### PR TITLE
Call tracing

### PR DESCRIPTION
**Pull request by hkalodner**

### CODE REVIEW

**Review by joshuacolvin0 — APPROVED** [review]
LGTM

File: packages/arb-avm-cpp/speedtest/speed_test.go [hunk 1]
```diff
@@ -62,6 +63,7 @@ func runExecutableFile(b *testing.B, filePath string) {
 	if err != nil {
 		b.Fatal(err)
 	}
+	defer runtime.KeepAlive(cursor)
```
Comment (hkalodner) [inline packages/arb-avm-cpp/speedtest/speed_test.go:66]: Similar to above, I'm confused why we need these outside of the direct cgo interaction

File: packages/arb-avm-cpp/cmachine/arbcore.go [hunk 2]
```diff
@@ -399,6 +402,23 @@ func (ac *ArbCore) AdvanceExecutionCursor(executionCursor core.ExecutionCursor,
 	return cursor.updateValues()
 }
 
+func (ac *ArbCore) AdvanceExecutionCursorWithTracing(executionCursor core.ExecutionCursor, maxGas *big.Int, goOverGas bool, allowSlowLookup bool, logNumber *big.Int) ([]value.Value, error) {
+	defer runtime.KeepAlive(ac)
```
Comment (hkalodner) [inline packages/arb-avm-cpp/cmachine/arbcore.go:406]: Might need a `defer runtime.KeepAlive(cursor)`

File: packages/arb-avm-cpp/cmachine/executioncursor_test.go [hunk 3]
```diff
@@ -54,6 +55,7 @@ func TestExecutionCursor(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	defer runtime.KeepAlive(cursor)
```
[the remark this answers is not in the document]
Comment (joshuacolvin0) [reply packages/arb-avm-cpp/cmachine/executioncursor_test.go:58]: :+1:

File: packages/arb-avm-cpp/cmachine/cmachine_test.go [hunk 4]
```diff
@@ -56,6 +57,7 @@ func TestMachineCreation(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	defer runtime.KeepAlive(cursor)
```
[the remark this answers is not in the document]
Comment (joshuacolvin0) [reply packages/arb-avm-cpp/cmachine/cmachine_test.go:60]: :+1:

File: packages/arb-avm-cpp/speedtest/speed_test.go (hunk 1 above)
Comment (joshuacolvin0) [reply packages/arb-avm-cpp/speedtest/speed_test.go]: :+1:

File: packages/arb-avm-cpp/cmachine/arbcore.go (hunk 2 above)
Comment (joshuacolvin0) [reply packages/arb-avm-cpp/cmachine/arbcore.go:406]: :+1:

**Review by joshuacolvin0 — DISMISSED** [review]
LGTM

**Review by joshuacolvin0 — DISMISSED** [review]
LGTM

**Review by joshuacolvin0 — APPROVED** [review]
LGTM

**Review by joshuacolvin0 — APPROVED** [review]
LGTM

**Review by joshuacolvin0 — DISMISSED** [review]
LGTM

**Review by joshuacolvin0 — APPROVED** [review]
LGTM

**Review by joshuacolvin0 — APPROVED** [review]
LGTM

**Review by joshuacolvin0 — APPROVED** [review]
LGTM

**Pull request merged**